flink-connectors/flink-connector-hbase/src/test/java/org/apache/flink/connector/hbase/source/HBaseSourceITCase.java

```diff
@@ -0,0 +1,313 @@
+
+/** Tests the most basic use cases of the source with a mocked HBase system. */
+public class HBaseSourceITCase extends TestsWithTestHBaseCluster {
+
+    private DataStream<String> streamFromHBaseSource(
+            StreamExecutionEnvironment environment, String tableName)
+            throws ParserConfigurationException, SAXException, IOException {
+        return streamFromHBaseSource(environment, tableName, 1);
+    }
+
+    private DataStream<String> streamFromHBaseSource(
+            StreamExecutionEnvironment environment, String tableName, int parallelism)
+            throws ParserConfigurationException, SAXException, IOException {
+        HBaseStringDeserializationScheme deserializationScheme =
+                new HBaseStringDeserializationScheme();
+        HBaseSource<String> source =
+                new HBaseSource<>(deserializationScheme, tableName, cluster.getConfig());
+        environment.setParallelism(parallelism);
+        DataStream<String> stream =
+                environment.fromSource(
+                        source,
+                        WatermarkStrategy.noWatermarks(),
+                        "hbaseSourceITCase",
+                        deserializationScheme.getProducedType());
+        return stream;
+    }
+
+    private static <T> void expectFirstValuesToBe(
+            DataStream<T> stream, T[] expectedValues, String message) {
+
+        List<T> collectedValues = new ArrayList<>();
+        stream.flatMap(
+                new RichFlatMapFunction<T, Object>() {
+
+                    @Override
+                    public void flatMap(T value, Collector<Object> out) {
+                        System.out.println("Test collected: " + value);
+                        collectedValues.add(value);
+                        if (collectedValues.size() == expectedValues.length) {
+                            assertArrayEquals(message, expectedValues, collectedValues.toArray());
+                            throw new SuccessException();
+                        }
+                    }
+                });
+    }
+
+    private static void doAndWaitForSuccess(
+            StreamExecutionEnvironment env, Runnable action, int timeout) {
+        try {
+            JobClient jobClient = env.executeAsync();
+            MiniCluster miniCluster = Util.miniCluster((MiniClusterJobClient) jobClient);
+            Util.waitForClusterStart(miniCluster, true);
+
+            action.run();
+            jobClient.getJobExecutionResult().get(timeout, TimeUnit.SECONDS);
+            jobClient.cancel();
+            throw new RuntimeException("Waiting for the correct data timed out");
+        } catch (Exception exception) {
+            if (!causedBySuccess(exception)) {
+                throw new RuntimeException("Test failed", exception);
+            } else {
+                // Normal termination
+            }
+        }
+    }
+
+    @Before
+    public void makeSignalFolder() {
+        makeFolder();
```

Comment:
Use `TemporaryFolder` rule?

Reply:
f6702d0e6bd16454bceac0d1a90091db8cb20f51 puts signal folder setup at least to test rule. 
Making the `FileSignal` class itself a rule would create the necessity to un-static-ify its methods, then the clean `signalSuccess()`-ish Syntax is lost to something like `signal.signalSuccess()`

Reply:
✅ 
Whole `FileSignal` was made a TestRule in d99a9f067c91ae9797deade194aed85fc835f9d1